internal/provider/data_source_packer_image.go

```diff
@@ -111,9 +122,49 @@ func dataSourcePackerImageRead(ctx context.Context, d *schema.ResourceData, meta
 
 	log.Printf("[INFO] Reading HCP Packer registry (%s) [project_id=%s, organization_id=%s, iteration_id=%s]", bucketName, loc.ProjectID, loc.OrganizationID, iterationID)
 
-	iteration, err := clients.GetIterationFromId(ctx, client, loc, bucketName, iterationID)
-	if err != nil {
-		return diag.FromErr(err)
+	var iteration *packermodels.HashicorpCloudPackerIteration
+	var err error
+
+	if iterID, ok := iterationID.(string); ok && iterID != "" {
+		iteration, err = clients.GetIterationFromId(
+			ctx,
+			client,
+			loc,
+			bucketName,
+			iterID)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	var channel *packermodels.HashicorpCloudPackerChannel
+
+	if chanSlug, ok := channelName.(string); ok && chanSlug != "" {
+		channel, err = clients.GetPackerChannelBySlug(
+			ctx,
+			client,
+			loc,
+			bucketName,
+			chanSlug)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	var diags diag.Diagnostics
```

Comment:
This variable is never being set

Reply:
True, that's an artifact from when both were allowed so long as there wasn't a conflict.
I'll remove this